docs/copilot/advanced.md

```diff
@@ -0,0 +1,84 @@
+
+* /help: Prints general help about GitHub Copilot.
+* /clear: Clear the session.
+
+## Creating a workspace
+
+You can use `@workspace /new` to scaffold a new project by using a natural language description:
+
+![Asking the @workspace agent to scaffold a new Node.js project using TypeScript](images/advanced/workspace-agent-new-example.png)
+
+The agent lists the directory structure for the new project. You can then create a new workspace for this project by selecting **Create Workspace**.
+
+Copilot Chat renders a file tree with the proposed directory structure in the chat response. You can select files to open a readonly preview in the editor. If Copilot's initial proposal wasn't quite right, you can ask follow-up questions to help Copilot iterate and improve.
+
+## Fixing code
+
+Several [slash commands](#agents-and-slash-commands) also work in inline chat such as `/explain` to ask Copilot to explain a block of code or `/tests` to generate unit tests.
+
+A useful command is `/fix`, which analyzes the error line or selected code and proposes a fix in a diff view for your review. After reviewing and possibly modifying the generated code, you can choose to **Accept** or **Discard** the fix.
+
+![Inline chat /fix identifying an unassigned variable and proposing a fix in a diff view](images/advanced/inline-chat-fix-example.png)
+
+If the cursor is on an error line, inline chat can suggest a fix for the error. For example, when you're missing a dependency, the chat response provides a command to install that dependency. You can then use the **Insert into terminal**  button or `kb(workbench.action.chat.runInTerminal) to insert the command in the terminal.
+
+![Inline chat /fix for an error proposing to install a missing library](images/advanced/inline-chat-fix-error-message-example.png)
+
+## Generating code documentation
+
+It can be tedious to add documentation comments to your code. The Copilot inline chat lets you use the `/doc` slash command to generate documentation for the entire code file or the selection. Copilot Chat supports multiple languages for generating code documentation.
+
+![Inline chat /doc example to generate documentation code comments for a calculator class](images/advanced/inline-chat-doc-example.png)
+
+## Generating unit tests
+
+The `@workspace /tests` slash command enables you to generate unit tests for your code. When generating the tests, the agent can detect the testing framework you're using and generates new tests in the same style. When you use inline chat, you can use the `/tests` slash command.
+
+![Chat view showing the /tests slash command to generate unit tests for the Divide method in a C# calculator class](images/advanced/workspace-agent-tests-example.png)
+
+When inline chat proposes creating new files, such as when using the `/tests` slash command, you can choose the file name and location by selecting **Create As** from the **Create** drop down.
```

Comment:
Should we add a section about getting help with failed terminal commands and cross references from here: https://github.com/ntrogh/vscode-docs/blob/c110423efc2b474d97bbc2c242cfc101e27d244a/docs/terminal/shell-integration.md#L115-L118